File: src/backend/routes/listenerdashboard.py
from flask import Blueprint, jsonify, render_template

listenerdashboard_bp = Blueprint("listenerdashboard_bp", __name__, url_prefix="/listener-dashboard")

@listenerdashboard_bp.route("/", methods=["GET"])
def listener_dashboard():
    """Render the Listener Intelligence Dashboard."""
    return render_template("listenerdashboard/listenerdashboard.html")

@listenerdashboard_bp.route("/api/engagement-metrics", methods=["GET"])
def get_engagement_metrics():
    # Mock data for engagement metrics
    data = {
        "dropOffPoints": [10, 20, 45],
        "relistens": 120,
        "shares": 50,
        "reviews": 30,
    }
    return jsonify(data)

@listenerdashboard_bp.route("/api/emotional-flow", methods=["GET"])
def get_emotional_flow():
    # Mock data for emotional flow
    data = {
        "episodes": [
            {"title": "Episode 1", "sentiment": [0.1, 0.5, -0.2]},
            {"title": "Episode 2", "sentiment": [0.3, 0.4, 0.2]},
        ]
    }
    return jsonify(data)

@listenerdashboard_bp.route("/api/cta-performance", methods=["GET"])
def get_cta_performance():
    # Mock data for CTA performance
    data = {
        "ctaClicks": 100,
        "conversionRate": 0.25,
    }
    return jsonify(data)

@listenerdashboard_bp.route("/api/superfan-identification", methods=["GET"])
def get_superfan_identification():
    # Mock data for superfans
    data = {
        "superfans": [
            {"name": "John Doe", "engagementScore": 95},
            {"name": "Jane Smith", "engagementScore": 90},
        ]
    }
    return jsonify(data)

@listenerdashboard_bp.route("/api/content-recommendations", methods=["GET"])
def get_content_recommendations():
    # Mock data for content recommendations
    data = {
        "recommendations": [
            {"title": "How to grow your audience", "category": "Marketing"},
            {"title": "Top 10 podcasting tools", "category": "Technology"},
        ]
    }
    return jsonify(data)

File: src/Frontend/static/js/listenerdashboard.js
import { showNotification } from "../podcastmanagement/podcastmanagement.js";

document.addEventListener("DOMContentLoaded", () => {
  console.log("Listener Intelligence Dashboard loaded");

  // Fetch and render engagement metrics
  fetch("/api/engagement-metrics")
    .then((response) => response.json())
    .then((data) => renderEngagementMetrics(data))
    .catch((error) => showNotification("Error", "Failed to load engagement metrics", "error"));

  // Fetch and render emotional flow mapping
  fetch("/api/emotional-flow")
    .then((response) => response.json())
    .then((data) => renderEmotionalFlow(data))
    .catch((error) => showNotification("Error", "Failed to load emotional flow data", "error"));

  // Fetch and render CTA performance
  fetch("/api/cta-performance")
    .then((response) => response.json())
    .then((data) => renderCtaPerformance(data))
    .catch((error) => showNotification("Error", "Failed to load CTA performance", "error"));

  // Fetch and render superfan identification
  fetch("/api/superfan-identification")
    .then((response) => response.json())
    .then((data) => renderSuperfanIdentification(data))
    .catch((error) => showNotification("Error", "Failed to load superfan data", "error"));

  // Fetch and render content recommendations
  fetch("/api/content-recommendations")
    .then((response) => response.json())
    .then((data) => renderContentRecommendations(data))
    .catch((error) => showNotification("Error", "Failed to load content recommendations", "error"));
});

function renderEngagementMetrics(data) {
  // Render engagement metrics chart
  console.log("Engagement Metrics:", data);
}

function renderEmotionalFlow(data) {
  // Render emotional flow heatmap
  console.log("Emotional Flow:", data);
}

function renderCtaPerformance(data) {
  // Render CTA performance table
  console.log("CTA Performance:", data);
}

function renderSuperfanIdentification(data) {
  // Render superfan list
  console.log("Superfan Identification:", data);
}

function renderContentRecommendations(data) {
  // Render content recommendations
  console.log("Content Recommendations:", data);
}
